src/ir.cpp

```diff
@@ -15677,6 +15745,895 @@ static TypeTableEntry *ir_analyze_instruction_offset_of(IrAnalyze *ira,
     return ira->codegen->builtin_types.entry_num_lit_int;
 }
 
+static void ensure_field_index(TypeTableEntry *type, const char *field_name, size_t index)
+{
+    Buf *field_name_buf;
+
+    assert(type != nullptr && !type_is_invalid(type));
+    // Check for our field by creating a buffer in place then using the comma operator to free it so that we don't
+    // leak memory in debug mode.
```

Comment:
This attention to detail is appreciated, but just so you know: the  c++ compiler doesn't even pretend to try to free memory. We treat the entire OS's memory as one big ArenaAllocator that gets freed upon process exit.

The self hosted compiler has a higher standard though. Because it is meant to be used as a long running process that performs multiple compilations, it will need to get these details correct.